shell/platform/windows/system_utils.cc

```diff
@@ -25,20 +26,48 @@ std::vector<LanguageInfo> GetPreferredLanguageInfo() {
 
 std::vector<std::wstring> GetPreferredLanguages() {
   std::vector<std::wstring> languages;
+  BOOL languages_from_registry = TRUE;
+  ULONG buffer_size = 0;
+  ULONG count = 0;
   DWORD flags = MUI_LANGUAGE_NAME | MUI_UI_FALLBACK;
 
-  // Get buffer length.
-  ULONG count = 0;
-  ULONG buffer_size = 0;
-  if (!::GetThreadPreferredUILanguages(flags, &count, nullptr, &buffer_size)) {
-    return languages;
+  // Determine where languages are defined and get buffer length
+  if (RegGetValueA(HKEY_CURRENT_USER,
+                   "Control panel\\International\\User Profile", "Languages",
+                   RRF_RT_REG_MULTI_SZ, NULL, NULL, &buffer_size) != S_OK) {
+    languages_from_registry = FALSE;
+    if (!::GetThreadPreferredUILanguages(flags, &count, nullptr,
+                                         &buffer_size)) {
+      return languages;
+    }
   }
 
-  // Get the list of null-separated languages.
+  // Mutli-string must be at least 3-long if non-empty,
+  // as a multi-string is terminated with 2 nulls.
+  //
+  // See: https://learn.microsoft.com/en-us/windows/win32/sysinfo/registry-value-types
```

Comment:
Microsoft docs are localized only if you remove the language code:

```suggestion
  // See: https://learn.microsoft.com/windows/win32/sysinfo/registry-value-types
```